src/main/java/com/netflix/playback/features/model/impl/PeriodAggregator.java

```diff
@@ -0,0 +1,111 @@
+package com.netflix.playback.features.model.impl;
+
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.google.common.collect.EvictingQueue;
+
+/**
+ * Based on various use cases, the aggregator return the data. It stores the events count for each period.
+ * 
+ */
+public class PeriodAggregator {
+
+	private final ConcurrentMap<Integer, ConcurrentMap<Integer, Integer>> periodToEventCount 
+				= new ConcurrentHashMap<>();
+	private ExecutorService executor = Executors.newFixedThreadPool(1);
+
+	// keep track of stats
+	private EvictingQueue<Integer> totalStatsQueue = EvictingQueue.create(2);
+	
+	// do a callback to aggregate data asynchronously
+	public void flush(ConcurrentMap<Integer, AtomicInteger> countAggregator, int currPeriod) {
+		FutureTask task = new FutureTask<>(new StatsCallable(countAggregator, currPeriod));
```

Comment:
we could have done it using Runnable interface and having StatsCallable as StatsRunnable implementing Runnable interface
https://github.com/hr1383/playback-features/pull/1